worker/crates/action-sink/src/file/tree.rs

```diff
@@ -0,0 +1,208 @@
+            child10: None,
+            child11: None,
+            contents: vec![],
+            min_lng: f64::MAX,
+            max_lng: f64::MIN,
+            min_lat: f64::MAX,
+            max_lat: f64::MIN,
+            min_height: f64::MAX,
+            max_height: f64::MIN,
+        }
+    }
+}
+
+impl Tile {
+    fn update_boundary(&mut self) {
+        for child in [
+            &mut self.child00,
+            &mut self.child01,
+            &mut self.child10,
+            &mut self.child11,
+        ]
+        .into_iter()
+        .flatten()
+        {
+            child.update_boundary();
+            self.min_lng = self.min_lng.min(child.min_lng);
+            self.max_lng = self.max_lng.max(child.max_lng);
+            self.min_lat = self.min_lat.min(child.min_lat);
+            self.max_lat = self.max_lat.max(child.max_lat);
+            self.min_height = self.min_height.min(child.min_height);
+            self.max_height = self.max_height.max(child.max_height);
+        }
+        for content in &self.contents {
+            self.min_lng = self.min_lng.min(content.min_lng);
+            self.max_lng = self.max_lng.max(content.max_lng);
+            self.min_lat = self.min_lat.min(content.min_lat);
+            self.max_lat = self.max_lat.max(content.max_lat);
+            self.min_height = self.min_height.min(content.min_height);
+            self.max_height = self.max_height.max(content.max_height);
+        }
+    }
+
+    fn into_tileset_tile(mut self) -> tileset::Tile {
+        self.update_boundary();
+
+        let children = {
+            let children: Vec<_> = [self.child00, self.child01, self.child10, self.child11]
+                .into_iter()
+                .flatten()
+                .map(|child| child.into_tileset_tile())
+                .collect();
+            if children.is_empty() {
+                None
+            } else {
+                Some(children)
+            }
+        };
+
+        let (content, contents) = {
+            match self.contents.len() {
+                0 => (None, None),
+                1 => {
+                    let content = tileset::Content {
+                        uri: self.contents[0].content_path.clone(),
+                        ..Default::default()
+                    };
+                    (Some(content), None)
+                }
+                _ => {
+                    let contents: Vec<_> = self
+                        .contents
+                        .into_iter()
+                        .map(|content| tileset::Content {
+                            uri: content.content_path,
+                            ..Default::default()
+                        })
+                        .collect();
+                    (None, Some(contents))
+                }
+            }
+        };
+
+        let (z, _, y) = self.zxy;
+        tileset::Tile {
+            geometric_error: geometric_error(z, y),
+            refine: Some(tileset::Refine::Replace),
```

Comment:
I would like Refine to be able to be set as a function argument.
The default is “replace”.

Reply:
Fixed